### ISSUE TITLE
<includeSystemCheckConstraints/> fails with missing FROM-clause entry for table "alias_18649163" on PostgreSQL

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10940]
There seems to have been a regression in recent jOOQ versions when turning on the following flag in PostgreSQL

```xml
<includeSystemCheckConstraints>true</includeSystemCheckConstraints>
```

The error is:

```
13:48:25,313 [WARNING] SQL exception            : Exception while executing meta query: ERROR: missing FROM-clause entry for table "alias_18649163"
  Position: 725
```

Logged SQL query:

```sql
select
  alias_122251260.nspname,
  alias_104821198.relname,
  pc.conname as constraint_name,
  replace(
    pg_get_constraintdef(pc.oid),
    'CHECK ',
    ''
  ) as check_clause
from (
  pg_catalog.pg_constraint as pc
    join (
      pg_catalog.pg_class as alias_104821198
        join pg_catalog.pg_namespace as alias_122251260
          on alias_104821198.relnamespace = alias_122251260.oid
    )
      on pc.conrelid = alias_104821198.oid
  )
where (
  alias_122251260.nspname in (
    'public', 'multi_schema', 'information_schema', 'pg_catalog', 's_5507', 's_7070_a',
    's_7070_b', 's_7271_d', 's_7271_r', 's_7274', 's_7496', 's_7685', 's_7495',
    's_10305_a'
  )
  and pc.contype = 'c'
)
union all
select
  alias_122251260.nspname,
  alias_104821198.relname,
  cc.constraint_name,
  cc.check_clause
from pg_catalog.pg_constraint as pc
  join information_schema.check_constraints as cc
    on (
      pc.conname = cc.constraint_name
      and alias_18649163.nspname = cc.constraint_name
    )
where (
  alias_18649163.nspname in (
    'public', 'multi_schema', 'information_schema', 'pg_catalog', 's_5507', 's_7070_a',
    's_7070_b', 's_7271_d', 's_7271_r', 's_7274', 's_7496', 's_7685', 's_7495',
    's_10305_a'
  )
  and (alias_122251260.nspname, alias_104821198.relname, cc.constraint_name) not in (
    select
      alias_122251260.nspname,
      alias_104821198.relname,
      pc.conname
    from (
      pg_catalog.pg_constraint as pc
        join (
          pg_catalog.pg_class as alias_104821198
            join pg_catalog.pg_namespace as alias_122251260
              on alias_104821198.relnamespace = alias_122251260.oid
        )
          on pc.conrelid = alias_104821198.oid
        join pg_catalog.pg_namespace as alias_18649163
          on pc.connamespace = alias_18649163.oid
      )
    where pc.contype = 'c'
  )
)
order by
  1,
  2,
  3
```

Perhaps related: https://github.com/jOOQ/jOOQ/issues/9961

**Comment by lukaseder** [comment]
A simpler case is documented here, including a workaround: #10992. We can apply the workaround to the jOOQ meta query prior to fixing this.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#10993)

**Comment by lukaseder** [comment]
The underlying problem has been fixed: https://github.com/jOOQ/jOOQ/issues/10992. With that, the workaround can be reverted
